gorm/transaction.go

```diff
@@ -173,10 +173,15 @@ func (t *Transaction) Commit(ctx context.Context) error {
 // Client is responsible to call `txn.Begin()` to open transaction.
 // If call of grpc.UnaryHandler returns with an error the transaction
 // is aborted, otherwise committed.
-func UnaryServerInterceptor(db *gorm.DB) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(db *gorm.DB, opts ...func(context.Context, interface{}, *Transaction)) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
 		// prepare new *Transaction instance
 		txn := &Transaction{parent: db}
+		if len(opts) > 0 {
+			for _, opt := range opts {
+				opt(ctx, req, txn)
+			}
+		}
```

Comment:
Please describe your use case, tests, and/or examples. It is not clear from this Pr what you are trying to accomplish. 

The functional option is very limited in scope. It would be better to implement flexible options where this callback behavior is just one type. See example below

```
func WithCallback(func(context.Context, interface{}, *Transaction) error) Option {
}

func UnaryServerInterceptor(db,  opts ...Option) grpc.UnaryServerInterceptor {
...
}
```

Reply:
use case PR messaged via DM.  Drew's function sig is much more flexible.